tools/generate_linux_syscalls.zig

```diff
@@ -22,702 +28,223 @@ const stdlib_renames = std.StaticStringMap([]const u8).initComptime(.{
     // ARM EABI/Thumb.
     .{ "arm_sync_file_range", "sync_file_range" },
     .{ "arm_fadvise64_64", "fadvise64_64" },
-    // ARC and Hexagon.
-    .{ "mmap_pgoff", "mmap2" },
-});
-
-// Only for newer architectures where we use the C preprocessor.
-const stdlib_renames_new = std.StaticStringMap([]const u8).initComptime(.{
-    .{ "newuname", "uname" },
-    .{ "umount", "umount2" },
 });
 
-// We use this to deal with the fact that multiple syscalls can be mapped to sys_ni_syscall.
-// Thankfully it's only 2 well-known syscalls in newer kernel ports at the moment.
-fn getOverridenNameNew(value: []const u8) ?[]const u8 {
-    if (mem.eql(u8, value, "18")) {
-        return "sys_lookup_dcookie";
-    } else if (mem.eql(u8, value, "42")) {
-        return "sys_nfsservctl";
-    } else {
-        return null;
-    }
-}
-
-fn isReservedNameOld(name: []const u8) bool {
-    return std.mem.startsWith(u8, name, "available") or
-        std.mem.startsWith(u8, name, "reserved") or
-        std.mem.startsWith(u8, name, "unused");
+/// Filter syscalls that aren't actually syscalls.
+fn isReserved(name: []const u8) bool {
+    return mem.startsWith(u8, name, "available") or
+        mem.startsWith(u8, name, "reserved") or
+        mem.startsWith(u8, name, "unused");
 }
 
-const default_args: []const []const u8 = &.{
-    "-E",
-    // -dM is cleaner, but -dD preserves iteration order.
-    "-dD",
-    // No need for line-markers.
-    "-P",
-    "-nostdinc",
-    // Using -I=[dir] includes the zig linux headers, which we don't want.
-    "-Itools/include",
-    "-Itools/include/uapi",
-    // Output the syscall in a format we can easily recognize.
-    "-D __SYSCALL(nr, nm)=zigsyscall nm nr",
+/// Values of the `abi` field in use by the syscall tables.
+///
+/// Since c. 2012, all new Linux architectures use the same numbers for their syscalls.
+/// Before kernel 6.11, the source of truth for this list was the arch-specific `uapi` headers.
+/// The 6.11 release converted this into a unified table with the same format as the older archs.
+/// For these targets, syscalls are enabled/disabled based on the `abi` field.
+/// These fields are sourced from the respective `arch/{arch}/kernel/Makefile.syscalls`
+/// files in the kernel source tree.
+/// Architecture-specific syscalls between [244...259] are also enabled by adding the arch name as an abi.
+const Abi = enum {
+    /// Syscalls common to two or more sub-targets.
+    /// Often used for single targets in lieu of a nil value.
+    common,
+    /// Syscalls using 64-bit types on 32-bit targets.
+    @"32",
```

Comment:
Is this comment correct? (Just making sure since it's a bit confusing given the name.)

Reply:
Yep. This is for time64 and largefile syscalls.